sqla_vertica_python/vertica_python.py

```diff
@@ -362,8 +362,25 @@ def get_check_constraints(self, connection, table_name, schema=None, **kw):
     # constraints are enforced on selects, but returning nothing for these
     # methods allows table introspection to work
 
-    def get_pk_constraint(self, bind, table_name, schema, **kw):
-        return {'constrained_columns': [], 'name': 'undefined'}
+    @reflection.cache
+    def get_pk_constraint(self, connection, table_name, schema=None, **kw):
+        query = None
+        query = None
+        if schema is not None:
+            query = "select constraint_id, constraint_name, column_name from v_catalog.constraint_columns \n\
+            WHERE table_name = '" + table_name + "' AND table_schema = '" + schema + "' \n\
+                  AND constraint_type = 'p'"
+        else:
+            query = "select constraint_id, constraint_name, column_name from v_catalog.constraint_columns \n\
+            WHERE table_name = '" + table_name + "' \n\
+                  AND constraint_type = 'p'"
+
+        rs = connection.execute(query)
+
+        cols = {row[2] for row in rs}
+        name = {row[1] for row in rs}
```

Comment:
Correct me if I'm wrong, but the `cols` declaration will exhaust the `rs` generator. You should instead do something like:

```python
cols = set()
name = set()
for row in connection.execute(query):
    name.add(row[1])
    cols.add(row[2])

Reply:
New commit implements suggestion.